src/Autodiscovery/AutodiscoveryLockResolver.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Goedemiddag\AutodiscoveryLock\Autodiscovery;
+
+use Illuminate\Support\Collection;
+
+final class AutodiscoveryLockResolver
+{
+    public const NOT_IN_LOCK = 'notInLock';
+    public const NOT_IN_AUTOLOAD = 'notInAutoload';
+    public function resolve(LaravelPackageManifest $manifest): Collection
+    {
+        $autoload = $manifest->collectManifestFromComposerAutoload();
+        $lock = $manifest->collectManifestFromLock();
+
+        return collect(
```

Comment:
The collection seems to be used as fancy associative array, maybe a value object could help here?